nexus/types/src/deployment/execution/utils.rs

```diff
@@ -0,0 +1,228 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::{
+    collections::{BTreeMap, HashMap},
+    net::{IpAddr, SocketAddrV6},
+};
+
+use internal_dns::{DnsConfigBuilder, ServiceName};
+use nexus_sled_agent_shared::inventory::SledRole;
+use omicron_common::{
+    address::{Ipv6Subnet, SLED_PREFIX},
+    api::external::Name,
+};
+use omicron_uuid_kinds::SledUuid;
+
+use crate::{
+    deployment::{
+        blueprint_zone_type, Blueprint, BlueprintZoneFilter, BlueprintZoneType,
+    },
+    internal_api::params::{DnsConfigZone, DnsRecord},
+    silo::{default_silo_name, silo_dns_name},
+};
+
+use super::Overridables;
+
+/// The minimal information needed to represent a sled in the context of
```

Comment:
Is the code in this file basically unchanged, just moved?

Reply:
Yes, it's pure code movement.